src/Raven.Client/Documents/Subscriptions/SubscriptionCriteria.cs

```diff
@@ -124,19 +152,60 @@ public override void ConvertToJavascript(JavascriptConversionContext context)
 
                 using (javascriptWriter.Operation(node))
                 {
-                    if (node.Expression == Parameter)
+                    if (node.Type == typeof(DateTime))
                     {
-                        javascriptWriter.Write("this.");
-                        javascriptWriter.Write(node.Member.Name);
+                        //match DateTime expressions like call.Started, user.DateOfBirth, etc
+                        if (node.Expression == Parameter)
+                        {
+                            //transle it to Date.parse(this.Started)
+                            javascriptWriter.Write($"Date.parse(this.{node.Member.Name})");
+                            return;
+
+                        }
+
+                        //match expression where DateTime object is nested, like order.ShipmentInfo.DeliveryDate
+                        if (node.Expression != null)
+                        {
+                            
+                            javascriptWriter.Write("Date.parse("); 
+                            context.Visitor.Visit(node.Expression); //visit inner expression (i.e order.ShipmentInfo)
+                            javascriptWriter.Write($".{node.Member.Name})"); 
+                            return;
+                        }
+
+                        //match DateTime.Now , DateTime.UtcNow, DateTime.Today
+                        switch (node.Member.Name)
+                        {
+                            case "Now":
+                                javascriptWriter.Write("Date.now()");
+                                break;
+                            case "UtcNow":
+                                var utc = DateTime.UtcNow;
```

Comment:
No, you are recording the time you are doing this, you need to record the time you actually invoked it.